src/main/java/me/botsko/prism/serializers/items/ItemStackSerializer.java

```diff
@@ -0,0 +1,295 @@
+                // This is silly
+                enchs[i] = ench.getKey().getKey().getKey() + ":" + ench.getValue();
+                i++;
+            }
+            data.enchs = enchs;
+        } else if (meta instanceof EnchantmentStorageMeta) {
+            final EnchantmentStorageMeta bookEnchantments = (EnchantmentStorageMeta) meta;
+            if (bookEnchantments.hasStoredEnchants()) {
+                if (bookEnchantments.getStoredEnchants().size() > 0) {
+                    final String[] enchs = new String[bookEnchantments.getStoredEnchants().size()];
+                    int i = 0;
+                    for (final Map.Entry<Enchantment, Integer> ench : bookEnchantments.getStoredEnchants().entrySet()) {
+                        // This is absolutely silly
+                        enchs[i] = ench.getKey().getKey().getKey() + ":" + ench.getValue();
+                        i++;
+                    }
+                    data.enchs = enchs;
+                }
+            }
+        }
+        if (meta instanceof FireworkEffectMeta) {
+            applyFireWorkMetaToActionData(meta, data);
+        }
+        if (meta instanceof BannerMeta) {
+            List<Pattern> patterns = ((BannerMeta) meta).getPatterns();
+            Map<String, String> stringyPatterns = new HashMap<>();
+            patterns.forEach(
+                    pattern -> stringyPatterns.put(pattern.getPattern().getIdentifier(), pattern.getColor().name()));
+            data.bannerMeta = stringyPatterns;
+        }
+        return data;
+    }
+
+    private static void applyFireWorkMetaToActionData(ItemMeta meta, ItemStackSerializer data) {
+        final FireworkEffectMeta fireworkMeta = (FireworkEffectMeta) meta;
+        if (fireworkMeta.hasEffect()) {
+            final FireworkEffect effect = fireworkMeta.getEffect();
+            if (effect != null) {
+                if (!effect.getColors().isEmpty()) {
+                    final int[] effectColors = new int[effect.getColors().size()];
+                    int i = 0;
+                    for (final Color effectColor : effect.getColors()) {
+                        effectColors[i] = effectColor.asRGB();
+                        i++;
+                    }
+                    data.effectColors = effectColors;
+                }
+
+                if (!effect.getFadeColors().isEmpty()) {
+                    final int[] fadeColors = new int[effect.getColors().size()];
+                    final int i = 0;
+                    for (final Color fadeColor : effect.getFadeColors()) {
+                        fadeColors[i] = fadeColor.asRGB();
+                    }
+                    data.fadeColors = fadeColors;
+                }
+                if (effect.hasFlicker()) {
+                    data.hasFlicker = true;
+                }
+                if (effect.hasTrail()) {
+                    data.hasTrail = true;
+                }
+            }
+        }
+    }
+
+    public ItemStack toBukkit() {
```

Comment:
Codacy found an issue: [The method 'toBukkit()' has an NPath complexity of 3276, current threshold is 200](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=6368169)